src/main/scala/edu/berkeley/ce/rockslicing/BlockVTK.scala

```diff
@@ -159,25 +158,40 @@ object BlockVTK {
     }
     offsetIterator(offsets, localOffsets.toSeq)
   }
+
+  /*
+   * FIXME: This is very inefficient and we probably want to use the right data
+   * structures to deal with duplicates for us.
+   */
+  private def distinctArrays(vertices: Seq[Array[Double]], previous: Seq[Array[Double]]=Nil): Seq[Array[Double]] = {
+    vertices match {
+      case Nil => previous
+      case vert+:verts =>
+        if (previous.exists(previousVertex => previousVertex sameElements vert)) {
+          distinctArrays(verts, previous)
+        } else {
+          distinctArrays(verts, vert+:previous)
+        }
+    }
+  }
 }
 
 /**
   * Simple data structure to contain data that represents a rock block that can be turned into vtk format
-  * @param Rock block 
   * @constructor Create a new rock block in format that can be turned into vtk by rockProcessor
   */
 case class BlockVTK(block: Block) {
   val orientedVertices = BlockVTK.orientVertices(block)
-  val tupleVertices = orientedVertices.values.flatten.toSeq.distinct
-  val connectivityMap = BlockVTK.connectivity(orientedVertices, tupleVertices)
-  val vertices = tupleVertices flatMap { t =>
-    List(t._1, t._2, t._3)
+  val arrayVertices = BlockVTK.distinctArrays(orientedVertices.values.flatten.toSeq)
```

Comment:
Sounds fair to me. We'll have a better idea of the implications of this once the `treeReduce` portion has been sped up - right now we haven't really tried this portion of the code on the full data set anyhow!